lib/classes/savings_account.rb

```diff
@@ -0,0 +1,28 @@
+require 'awesome_print'
+module BankAccount
+  class Savings < Account
+
+    def initialize(id, initial_balance)
+      super
+      if @current_balance < 10
+        raise ArgumentError.new, "Initial deposit must be greater than 10"
+      end
+    end
+
+    def withdraw(amount)
+      super
+      if amount > (@current_balance - 10)
+        raise ArgumentError.new, "Savings cannot go below 10"
+      end
+      return (@current_balance -= 2)
+    end
+
+    def add_interest(rate)
```

Comment:
It seems like all of these variables are set as instance variables unnecessarily. I don't think `@rate` and `@interest` are needed anywhere else outside of this method.